### ISSUE TITLE
[instagram-feed] Add Videos and Image Gallery to feed

### ISSUE THREAD

**Issue by bjoernbohr** [getgrav/grav-premium-issues#391]
Hello,
i think this is more like a Feature request than an issue but our customer wants to add the Gallery images and Videos he uploaded to his feed. A simple link to media_url ist not possbile because of the expired urls. Is there a workaround to get these two media types working with the plugin or are there ideas to add the possibility to allow videos and gallery to be saved also so they can be loaded inside of the cms context?

**Comment by rhukster** [comment]
Videos are a bit more tricky than images.  For images, we are caching the images locally, getting around the timeouts and expired URLs by caching the images locally.  This is fine because images are quite small, and Grav's built in media handling can handle the expiration etc.  

Videos on the other hand are large, and require web server optimizations to ensure they are 'streamed' properly.  Caching the videos locally is not really feasible because the files are large, and downloading them directly could be limited by Facebook (probably is).  This means that if it's possible at all, it's going to be very slow to 'cache' them and then they are going to take up a lot of room. Then even if we could cache them locally, serving them would be sub-optimal compared to the highly optimized video serving you get with Instagram.

Basically the short answer is no, it's not really practical to integrate video in the same way we do images.  Instagram really want you going to Instagram to view the content, they don't make it easy, and this goes double for videos.

**Comment by bjoernbohr** [comment]
Yeah i figured as much, thanks for you reply

**Comment by rhukster** [comment]
Closed due to inactivity
